packages/next-app/src/app/pool/[poolId]/(components)/TransactionDialog.tsx

```diff
@@ -0,0 +1,225 @@
+"use client";
+import * as Alert from "@radix-ui/react-alert-dialog";
+import cn from "classnames";
+import { useContext, useEffect, useState } from "react";
+import {
+  ImCheckboxChecked,
+  ImCheckboxUnchecked,
+  ImHourGlass,
+  ImWarning,
+} from "react-icons/im";
+import useSWR from "swr";
+
+import { Button } from "#/components";
+import {
+  PoolMetadataAttribute,
+  PoolMetadataContext,
+  UpdateStatus,
+} from "#/contexts/PoolMetadataContext";
+import { pinJSON } from "#/lib/ipfs";
+import metadataGql from "#/lib/poolMetadataGql";
+import { fetcher } from "#/utils/fetcher";
+import { writeSetPoolMetadata } from "#/wagmi/setPoolMetadata";
+
+enum ButtonLabels {
+  "Waiting...",
+  "Confirm transaction on your wallet",
+  "Submiting...",
+  "Close!",
+  "Close",
+}
+
+const ActionStage = ({
+  description,
+  stage,
+  error,
+}: {
+  description: string;
+  stage: number;
+  error: number;
+}) => {
+  const { updateStatus } = useContext(PoolMetadataContext);
+
+  const Icon =
+    error === stage
+      ? ImWarning
+      : updateStatus < stage
+      ? ImCheckboxUnchecked
+      : updateStatus === stage
+      ? ImHourGlass
+      : ImCheckboxChecked;
+
+  return (
+    <div className="flex items-center space-x-2 text-xl">
+      <>
+        <Icon
+          className={`${
+            stage === error
+              ? "text-red-400"
+              : stage >= updateStatus
+              ? "text-white"
+              : "text-blue-500"
+          }`}
+        />
+        <div
+          className={`font-light 
+          
+          ${
+            stage === error
+              ? "text-red-400"
+              : stage > updateStatus
+              ? "text-slate-400"
+              : "text-white"
+          }`}
+        >
+          {description}
+        </div>
+      </>
+    </div>
+  );
+};
+
+export function TransactionDialog({
+  children,
+  poolId,
+}: React.PropsWithChildren<{
+  poolId: `0x${string}`;
+}>) {
+  const [open, setOpen] = useState(false);
+  const [stageError, setStageError] = useState(-1);
+
+  const {
+    metadata,
+    handleSetMetadata,
+    setStatus,
+    updateStatus,
+    submit,
+    handleSubmit,
+  } = useContext(PoolMetadataContext);
+
+  const { data: poolsData } = metadataGql.useMetadataPool({
+    poolId,
+  });
+
+  const pool = poolsData?.pools[0];
+  const { data } = useSWR(
+    pool?.metadataCID
+      ? `https://gateway.pinata.cloud/ipfs/${pool.metadataCID}`
```

Comment:
can we set this base URL as constant somewhere? I think we repeat it in another file

Reply:
I cleaned up the old pages. Now, we only use it here.